scripts/type-generation/generate-typings-utils.js

```diff
@@ -16,16 +16,17 @@ const changeType = (text) => {
     default: {
         text = text.toLowerCase()
     }
+    }
 
+    if (text.indexOf('promise.<') === 0) {
```

Comment:
please don't do any promise related changes here

Reply:
Mhm, regardless of the outcome of sync/async discussion, I believe this code is important to make Typescript typing work for promises (any of them, really).

Why do you think I should not change anything here?

Reply:
@CrispusDH this question is still relevant, not sure why this function should not be touched

Reply:
As we have already discussed, you should not do any changes in generate typings function

Reply:
HI @mgrybyk 

We indeed have discussed that typings should be changed and splitter between the two packages, but we haven't touched the topic of *why* code that generates incorrect Typescript typings should not be touched.

I believe you might be confused with the purpose of this change, it is partially described here: https://github.com/webdriverio/webdriverio/pull/3957/commits/266bedde2efc99353d42982b515a6e9376a94d37

At the moment our automatic typing generation scripts will generate `promise.<string>` type for Promises, which is not correct way of defining promise type in Typescript. The issue comes from the libraries we depend on (see commit message) and looks like it is an intentional behaviour based on fixtures.

For us to be able to generate correct typings for functions that return promises we need the change I've made here.

Could you please explain *why* should I not change this code?

Thanks!

Reply:
Can you please provide an example? 
Secondly, we can have another PR to fix typings related issues, this one is about waitUntil, and there is nothing to be changed in typing generation for it.

Reply:
Yes, of course, as I said this function without my fix will generate incorrect Promise typing like this: `Promise.<script>` (notice dot in the middle).
Feel free to switch to my branch, revert this particular change and run `npm run generate:typings`. Check the generated `d.ts` file and how it has `Promise.<T>` return types.

Otherwise, in master branch of this repository change return type for any function in jsdoc to `Promise<T>` and see the output in the generated `d.ts` file.

Change in how we are handling type generation is required for fixing `waitUntil()` typing issue. I don't think your suggestion to split it into 2 pull request is a good idea as it will create sequential interdependency between the PRs.

Both commit message and PR description are transparent on this change and the motivation for it.

Hope that explains the issue.

Reply:
I can't see such problem, can you please provide an example? Which commands have such issue?

Reply:
@mgrybyk I think I've exhausted my ability to explain the issue.  If you can't see the problem feel free to update this PR as required or decline it otherwise.

Reply:
Specific example would be really helpful, sorry for exhausting :+1: